service/src/main/java/tds/assessment/services/impl/AssessmentAssembler.java

```diff
@@ -3,57 +3,159 @@
+    }
+
+    /**
+     * Add content (other than {@link tds.assessment.Segment}s) to the {@link tds.assessment.Assessment} that is being
+     * assembled.
+     *
+     * @param assessment The assessment to update
+     * @param constraints The collection of {@link tds.assessment.ItemConstraint}s
+     * @param strands The collection of {@link tds.assessment.Strand}s that are associated to the assessment
+     * @param itemProperties The properties that are associated to each item
+     */
+    private void addAssessmentContent(Assessment assessment,
+                                      List<ItemConstraint> constraints,
+                                      Set<Strand> strands,
+                                      List<ItemProperty> itemProperties) {
         assessment.setItemConstraints(constraints);
         assessment.setStrands(strands.stream()
-                .filter(strand -> strand.getSegmentKey().equals(assessment.getKey())).collect(Collectors.toSet()));
-
-        for (Segment segment : assessment.getSegments()) {
-            segment.setForms(forms.stream()
-                    .filter(form -> form.getSegmentKey().equals(segment.getKey())).collect(Collectors.toList()));
-            segment.setItems(items.stream()
-                    .filter(item -> item.getSegmentKey().equals(segment.getKey())).collect(Collectors.toList()));
-            segment.setStrands(strands.stream()
-                    .filter(strand -> strand.getSegmentKey().equals(segment.getKey())).collect(Collectors.toSet()));
-        }
+            .filter(strand -> strand.getSegmentKey().equals(assessment.getKey()))
+            .collect(Collectors.toSet()));
+
+        // To determine which language(s) are available to an assessment, we need to look at the item properties for
+        // each item.  Each item should have a property with a name equal to "Language" (case-insensitive).  While
+        // iterating through the item properties, we look for the language item property and collect its value.
+        Set<String> languages = itemProperties.stream()
+            .filter(ip -> ip.getName().equalsIgnoreCase("language"))
+            .map(ItemProperty::getValue)
+            .collect(Collectors.toSet());
+        assessment.setLanguages(languages);
     }
 
+    /**
+     * Add the {@link tds.assessment.ItemProperty} instances to the appropriate {@link tds.assessment.Item}.
+     *
+     * @param items The items that need to be assigned item properties
+     * @param itemProperties The item properties that need to be mapped to their items
+     */
+    private void addItemPropertiesToItems(List<Item> items, List<ItemProperty> itemProperties) {
+        if (itemIdToPropertyMap == null) {
+            itemIdToPropertyMap = ArrayListMultimap.create();
+            itemProperties.stream()
+                .filter(property -> property.getItemId() != null)
+                .forEach(property -> itemIdToPropertyMap.put(property.getItemId(), property));
```

Comment:
I would need to play with it but it feels like a collect using collectors [groupingBy](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#groupingBy-java.util.function.Function-java.util.function.Supplier-java.util.stream.Collector- ) could solve this without having to mutate the outer map in a stream.  Added benefit is we could remove the ArrayListMultiMap